vms/platformvm/txs/executor/state_changes.go

```diff
@@ -257,10 +251,97 @@ func advanceTimeTo(
 		changes.DeleteExpiry(expiry)
 	}
 
+	if !backend.Config.UpgradeConfig.IsEtnaActivated(newChainTime) {
+		changes.SetTimestamp(newChainTime)
+		return changes, changed, nil
```

Comment:
Can/should this early return case be moved up above checking the expiries?

Reply:
It can be... I did it this way to avoid needing to modify tests (as unfortunately this part of the code has heavily used mock testing... So non-functional changes can require test modifications.